agenta-backend/agenta_backend/routers/app_router.py

```diff
@@ -594,8 +594,8 @@ async def create_app_and_variant_from_template(
 
             envvars = {**(payload.env_vars or {})}
             for key in supported_llm_prodviders_keys:
```

Comment:
So, the logic here should be:
Go over the providers, if the user provides a key, then use it, otherwise use our key.

Reply:
Got it.